### PR TITLE
Configurable ingress and support for newer versions of k8s

### PR DESCRIPTION
**Pull request by PashCracken**
- Make ingress configurable
- Update ingress apiVersion for newer k8s version
- Add labels in ingress
- Change deployment apiVersion to apps/v1 to support newer k8s versions

### CODE REVIEW

**Comment by salaboy** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **salaboy**

If they are not already assigned, you can assign the PR to them by writing `/assign @salaboy` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=zeebe-io%2Fzeebe-operate-helm).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/zeebe-io/zeebe-operate-helm/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by salaboy** [comment]
Hi @PashCracken. Thanks for your PR.

I'm waiting for a [zeebe-io](https://github.com/orgs/zeebe-io/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/zeebe-io/people) to skip this step.

Once the patch is verified, the new status will be reflected by the `ok-to-test` label.

I understand the commands that are listed [here](https://go.k8s.io/bot-commands?repo=zeebe-io%2Fzeebe-operate-helm).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by salaboy** [comment]
@PashCracken I think that this changes will break compatibility with older versions of Kube right? 
The change in the `deployment apiVersion to apps/v1` will mess things up for older versions.. I wonder if we can use a conditional to support both use cases

**Comment by salaboy** [comment]
@PashCracken you are doing it already for the ingress:
```
{{- if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
 apiVersion: networking.k8s.io/v1beta1
 {{- else -}}
 apiVersion: extensions/v1beta1
 {{- end }}

```

File: zeebe-operate/templates/deployment.yaml
Comment (salaboy) [inline zeebe-operate/templates/deployment.yaml:1]: @PashCracken can you do the same check here?

File: zeebe-operate/templates/ingress.yaml
Comment (salaboy) [inline zeebe-operate/templates/ingress.yaml:10]: @PashCracken this are all great changes.. and it great that it is coming from the community!

**Review by salaboy — CHANGES_REQUESTED** [review]
Can you please apply the requested changes?

File: zeebe-operate/templates/deployment.yaml
Comment (PashCracken) [reply zeebe-operate/templates/deployment.yaml:1]: @salaboy I’ll add the appropriate check on Monday

Comment (PashCracken) [reply zeebe-operate/templates/deployment.yaml:1]: @salaboy Deployment apps/v1 has been available since Kubernetes version 1.9. Do you really want to support versions older than that? If not, these changes will work fine for version 1.9 and newer

Comment (PashCracken) [reply zeebe-operate/templates/deployment.yaml]: @salaboy Deployment apps/v1 has been available since Kubernetes version 1.9. Do you really want to support versions older than that? If not, these changes will work fine for version 1.9 and newer

Comment (salaboy) [reply zeebe-operate/templates/deployment.yaml:1]: @PashCracken.. you are right.. I was checking that exactly.. I don't think that there is need for that.. I will merge and test the new versions with 1.14

**Pull request merged**
